### PR TITLE
Net core support & Xamarin consumption test

### PR DESCRIPTION
**Pull request by rhwilburn**
netstandard 98% API of net45 but 2% hasn't been migrated as it can be implemented by the end user and isn't supported by .net by default anymore. Those parts are described in the readme changes (ie service point manager)

Added a Xamarin project to validate that xamarin works and demonstrate how it works to consumers.

Updated the readme to include netcore information and supported frameworks info.

Removed TaskFactoryExtensions as it had no references and can be done by the end user. The end user can just call thread.sleep on what they are doing much more easily.

https://github.com/maximn/google-maps/issues/94

### CODE REVIEW

**Comment by maximn** [comment]
What is the `Components` folder?

**Comment by maximn** [comment]
`XamarinCanConsumeNetStandard`, maybe rename to something like `XamarinSampleApp`

**Comment by maximn** [comment]
too many changes in this one commit :(

File: GoogleMapsApi/Engine/MapsAPIGenericEngine.cs [hunk 1]
```diff
@@ -20,6 +20,8 @@ public abstract class MapsAPIGenericEngine<TRequest, TResponse>
         internal static event UriCreatedDelegate OnUriCreated;
         internal static event RawResponseReciviedDelegate OnRawResponseRecivied;
 
+#if NET45
```
Comment (maximn) [inline GoogleMapsApi/Engine/MapsAPIGenericEngine.cs]: hmmm .... Let's think if there's a nicer way?
What about the proposed way here- https://docs.microsoft.com/en-us/dotnet/core/porting/project-structure ?

**Comment by maximn** [comment]
@rhwilburn are you still here? any thoughts about my comments?

**Comment by rhwilburn** [comment]
@maximn Sounds good. I'm very limited in time in the next few months, so will do what I can.

**Comment by maximn** [comment]
@rhwilburn closing it for now. Let me know if you want to continue.

**Pull request closed without merging**